OpenSimRT/Common/include/CircularBuffer.h

```diff
@@ -38,7 +37,29 @@
 namespace OpenSimRT {
 
 /**
- * \brief A thread safe circular buffer.
+ * \brief A thread safe circular buffer. It is based on a circular array with
+ *  size equal to `history` and elements of type defined by the template
+ *  parameter T. New data are appended to the buffer using the `add()` function.
+ *  Older data are discarded when adding new data when the buffer is full.
+ *  Retrieving data from the buffer is performed using the `get(M)` function,
+ *  where M \in [1, history] is the number of the latest (newer) elements
+ *  inserted in the buffer. Retrieval is possible only if new elements have been
+ *  appended to the buffer. To retrieve data from the buffer (and terminate the
+ *  'wait' state of the consumer thread) even if no new data have been added,
+ *  the user must first call the `externalNotify()` function, which notifies the
```

Comment:
Try to give an example of the thread-safety and the externalNotify through code after the diagram, so it is clearer. It is not very clear what you mean "to retrieve data from the buffer even if no new data have been added.

Also, do you think externalNotify is a good name? How does it work?